features/admin.remote-userstores.v1/components/edit/userstore-general-settings.tsx

```diff
@@ -517,354 +516,356 @@ export const UserStoreGeneralSettings: FunctionComponent<UserStoreGeneralSetting
+                    assertionType="checkbox"
+                    primaryAction={ t("common:confirm") }
+                    secondaryAction={ t("common:cancel") }
+                    onSecondaryActionClick={ (): void => {
+                        setShowRegenerateConfirmationModal(false);
+                        setAlert(null);
+                    } }
+                    onPrimaryActionClick={
+                        (): void => handleRegenerateAgentConnectionToken(regeneratingAgentConnection)
+                    }
+                    closeOnDimmerClick={ false }
+                >
+                    <ConfirmationModal.Header data-testid={ `${componentId}-confirmation-modal-header` }>
+                        { t("user:deleteUser.confirmationModal.header") }
+                    </ConfirmationModal.Header>
+                    <ConfirmationModal.Message
+                        data-testid={ `${componentId}-confirmation-modal-message` }
+                        attached
+                        warning
                     >
-                        <DangerZone
-                            actionTitle={ t("userstores:dangerZone.disable.actionTitle") }
-                            header="Disable User Store"
-                            subheader={ t("userstores:dangerZone.disable.subheader") }
-                            onActionClick={ undefined }
-                            data-testid={ `${ testId }-delete-danger-zone` }
-                            toggle={ {
-                                checked: isDisabled !== undefined
-                                    ? isDisabled === "true"
-                                    : userStore?.properties?.find(
-                                        (property: UserStoreProperty) =>
-                                            property.name === DISABLED)?.value === "false",
-                                onChange: handleUserStoreDisable
-                            } }
-                        />
-                        <DangerZone
-                            actionTitle={ t("userstores:dangerZone.delete.actionTitle") }
-                            header={ t("userstores:dangerZone.delete.header") }
-                            subheader={ t("userstores:dangerZone.delete.subheader") }
-                            onActionClick={ () => setShowDeleteConfirmationModal(true) }
-                            data-testid={ `${ testId }-delete-danger-zone` }
-                        />
-                    </DangerZoneGroup>
-                )
-            }
+                        This action is irreversible and will revoke the previously used installation token.
+                    </ConfirmationModal.Message>
+                    <ConfirmationModal.Content>
+                        <div className="modal-alert-wrapper"> { alert && alertComponent }</div>
+                        If you regenerate the token the users with the previous token will no longer be authorised to
```

Comment:
Let's add i18n translations here

Reply:
These are already available. The component is separated out.